test/e2e/csv_e2e_test.go

```diff
@@ -421,4043 +273,3914 @@ var _ = Describe("ClusterServiceVersion", func() {
-			Expect(ps.Items).To(Not(BeEmpty()))
+			It("remains in phase Succeeded when only one pod is available", func() {
+				Eventually(func() int32 {
+					dep, err := c.GetDeployment(ns.GetName(), "deployment")
+					if err != nil || dep == nil {
+						return 0
+					}
+					return dep.Status.ReadyReplicas
+				}).Should(Equal(int32(2)))
+
+				var ps corev1.PodList
+				Expect(ctx.Ctx().Client().List(context.Background(), &ps, client.MatchingLabels{"app": "foobar"})).To(Succeed())
+				Expect(ps.Items).To(Not(BeEmpty()))
 
-			Expect(ctx.Ctx().Client().Delete(context.Background(), &ps.Items[0])).To(Succeed())
+				Expect(ctx.Ctx().Client().Delete(context.Background(), &ps.Items[0])).To(Succeed())
 
-			Consistently(func() (*operatorsv1alpha1.ClusterServiceVersion, error) {
-				return &csv, ctx.Ctx().Client().Get(context.Background(), client.ObjectKeyFromObject(&csv), &csv)
-			}).Should(HavePhase(operatorsv1alpha1.CSVPhaseSucceeded))
+				Consistently(func() (*operatorsv1alpha1.ClusterServiceVersion, error) {
+					return &csv, ctx.Ctx().Client().Get(context.Background(), client.ObjectKeyFromObject(&csv), &csv)
+				}).Should(CSVHasPhase(operatorsv1alpha1.CSVPhaseSucceeded))
+			})
 		})
-	})
+		When("a copied csv exists", func() {
+			var (
+				target   corev1.Namespace
+				original operatorsv1alpha1.ClusterServiceVersion
+				copyCSV  operatorsv1alpha1.ClusterServiceVersion
+			)
+
+			BeforeEach(func() {
+				target = corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "watched-",
+					},
+				}
+				Expect(ctx.Ctx().Client().Create(context.Background(), &target)).To(Succeed())
+
+				original = operatorsv1alpha1.ClusterServiceVersion{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       operatorsv1alpha1.ClusterServiceVersionKind,
+						APIVersion: operatorsv1alpha1.ClusterServiceVersionAPIVersion,
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "csv-",
+						Namespace:    ns.GetName(),
+					},
+					Spec: operatorsv1alpha1.ClusterServiceVersionSpec{
+						InstallStrategy: newNginxInstallStrategy(genName("csv-"), nil, nil),
+						InstallModes: []operatorsv1alpha1.InstallMode{
+							{
+								Type:      operatorsv1alpha1.InstallModeTypeAllNamespaces,
+								Supported: true,
+							},
+						},
+					},
+				}
+				Expect(ctx.Ctx().Client().Create(context.Background(), &original)).To(Succeed())
+
+				Eventually(func() error {
+					key := client.ObjectKeyFromObject(&original)
+					key.Namespace = target.GetName()
+					return ctx.Ctx().Client().Get(context.Background(), key, &copyCSV)
+				}).Should(Succeed())
+			})
 
-	When("a copied csv exists", func() {
-		var (
-			target   corev1.Namespace
-			original operatorsv1alpha1.ClusterServiceVersion
-			copy     operatorsv1alpha1.ClusterServiceVersion
-		)
+			AfterEach(func() {
+				if target.GetName() != "" {
+					Expect(ctx.Ctx().Client().Delete(context.Background(), &target)).To(Succeed())
```

Comment:
Do we need to use the TeardownNamespace helper function so we can also gather artifacts if this test case fails?

Reply:
(Updated in the last force push)